rasa/engine/recipes/graph_recipe.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+from rasa.engine.recipes.recipe import Recipe
+from rasa.engine.graph import GraphModelConfiguration
+from rasa.shared.constants import DOCS_URL_GRAPH_RECIPE
+from rasa.shared.data import TrainingType
+from rasa.shared.utils.common import mark_as_experimental_feature
+from rasa.shared.utils.io import raise_warning
+from rasa.engine.graph import GraphSchema
+from rasa.nlu.classifiers.regex_message_handler import RegexMessageHandler
+
+from typing import Dict, Text, Any
+
+
+logger = logging.getLogger(__name__)
+
+
+class GraphV1Recipe(Recipe):
+    """Recipe which converts the graph model config to train and predict graph."""
+
+    name = "graph.v1"
+
+    def graph_config_for_recipe(
+        self,
+        config: Dict,
+        cli_parameters: Dict[Text, Any],
+        training_type: TrainingType = TrainingType.BOTH,
```

Comment:
hmm - how is this determined? Maybe it could be based on what targets are provided in the config?

Reply:
It is determined from CLI argument, ie. `rasa train core` or `rasa train nlu`. If neither `core` nor `nlu` is provided, in that case it is "inferred", TrainingType.BOTH by default, but if stories or nlu files are missing then it is changed to the other. Because of this, I think training type is independent of targets, and maybe later on we can let users customize targets as well.